sqlglot/parser.py

```diff
@@ -226,6 +225,22 @@ def raise_error(self, message, token=None):
         if self.error_level == ErrorLevel.WARN:
             logging.error(self.error)
 
+    def expression(self, exp_class, **kwargs):
+        instance = exp_class(**kwargs)
+        self.validate(instance)
+        return instance
+
+    def validate(self, expression):
+        if self.error_level == ErrorLevel.IGNORE:
+            return
+
+        for k in expression.args:
+            if k not in expression.arg_types:
+                self.raise_error(f"Unexpected keyword: '{k}' for {expression.token_type}")
+        for k, v in expression.arg_types.items():
+            if v and expression.args.get(k) is None:
```

Comment:
This is something I realized after I merged my PR: should we also check for empty list here and not just None?

Reply:
not sure about this one, need to double check, it breaks some stuff

Reply:
hm i think it might work